<launch> 
  <!--include file="$(find vp6242_arm_navigation)/launch/vp6242_planning_environment.launch" /-->
  <param name="robot_description" textfile="$(find vp6242_description)/robot/vp6242.urdf" />
  <rosparam command="load" ns="robot_description_planning" file="$(find vp6242_arm_navigation)/config/vp6242_planning_description.yaml" />

  <!--include file="$(find planning_environment)/launch/planning_environment_visualization_prerequisites.launch" /-->
  <!--aram name="/use_sim_time" value="true" /-->

  <!--node pkg="planning_environment" name="wall_clock_server" type="fake_time.py" /-->

  <include file="$(find planning_environment)/launch/environment_server.launch" >
    <arg name="use_monitor" value="true"/>
    <arg name="use_collision_map" value="false"/>
  </include>
  
  <include file="$(find vp6242_arm_navigation)/launch/constraint_aware_kinematics.launch" />
  <include file="$(find vp6242_arm_navigation)/launch/ompl_planning.launch" />
  <include file="$(find vp6242_arm_navigation)/launch/trajectory_filter_server.launch" />


  <!-- RC7M -->
  <node name="vp6242" pkg="hg_cpp" type="hg_ros" output="screen">
    <param name="simulate" value="false"/>
    <param name="loop_rate" value="100.0"/>
    <param name="joint_publish_rate" value="50.0"/>
    <param name="diagnostic_publish_rate" value="10.0"/>      
    <rosparam file="$(find rc7m)/parameters/vp6242.yaml" command="load" />    
  </node>
  
  <node pkg="robot_state_publisher" type="state_publisher" name="rob_st_pub" output="screen">
      <!--remap from="robot_description" to="/arm1/robot_description" /-->
      <!--remap from="joint_states" to="/arm1/joint_states" /-->
  </node>
  
  <!-- fk/ik solver http://www.ros.org/wiki/arm_kinematics -->
  <!--node pkg="arm_kinematics" type="arm_kinematics" name="arm_kinematics" output="screen">
    <param name="root_name" value="base_link" />
    <param name="tip_name" value="link5" />    
  </node-->
  
  <node pkg="prw" type="personal_robotic_workspace" name="prw" output="screen">
    <!--remap from="robot_description" to="/arm1/robot_description" /-->    
  </node>
   
  <!--node pkg="tf" type="static_transform_publisher" name="table_broadcaster" 
  args="-0.8013312048757147 -0.4705600194028853 1.879842401563593 1.580096886 -0.16322015961780161 3.210146698 kinect7051A_rgb_optical_frame table 100" /-->
  
  <!--node pkg="tf" type="static_transform_publisher" name="arm1_broadcaster" 
  args="0.10 0.67 0.00 0 0 0 world base_link 100" /-->
  
  	
  <node pkg="rviz" type="rviz" name="rviz" output="screen" args="-d $(find prw)/cfg/prw.vcg">    
  </node>
  
  <param name="number_of_arm" value="2"/>  
  <param name="arm_group_0" value="arm" />
  <param name="arm_controller_0" value="/vp6242/follow_joint_trajectory" />
  <param name="ik_link_0" value="link5" />
  <param name="ik_name_0" value="/vp6242_arm_kinematics/get_constraint_aware_ik" />
  <param name="non_collision_ik_name_0" value="/vp6242_arm_kinematics/get_ik" />
  
  <param name="planner_service_name" value="/ompl_planning/plan_kinematic_path" />
  <param name="trajectory_filter_service_name" value="/trajectory_filter_server/filter_trajectory_with_constraints" />
  <param name="set_planning_scene_diff_name" value="environment_server/set_planning_scene_diff"/>
  <param name="visualizer_topic_name_" value="personal_robotic_workspace_visualizer_markers"/>
  
  
  
    
</launch>
